beluga/include/beluga/storage.hpp

```diff
@@ -179,10 +179,11 @@ class StoragePolicy : public Mixin {
   void initialize_particles(Range&& input) {
     static_assert(std::is_same_v<particle_type, ranges::range_value_t<Range>>, "Invalid value type");
     const std::size_t size = this->self().max_samples();
-    particles_.resize(size);
-    const auto first = std::begin(views::all(particles_));
+    particles_[1].resize(size);
+    const auto first = std::begin(views::all(particles_[1]));
     const auto last = ranges::copy(input | ranges::views::take(size), first).out;
```

Comment:
This is one of the knots of the problematic execution path. The other two are [here](https://github.com/Ekumen-OS/beluga/blob/31445270321482c72dc8a7608385a0682880151b/beluga/include/beluga/algorithm/sampling.hpp#L381) and [here](https://github.com/Ekumen-OS/beluga/blob/31445270321482c72dc8a7608385a0682880151b/beluga/include/beluga/algorithm/sampling.hpp#L162). Sampling an existing state means reading from the same `particles_` we are writing to.